flink-ml-lib/src/main/java/org/apache/flink/ml/classification/knn/KnnModel.java

```diff
@@ -0,0 +1,223 @@
+                                    outputTypeInfo);
+                        });
+        return new Table[] {tEnv.fromDataStream(output)};
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return this.params;
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        ReadWriteUtils.saveMetadata(this, path);
+        ReadWriteUtils.saveModelData(
+                KnnModelData.getModelDataStream(modelDataTable),
+                path,
+                new KnnModelData.ModelDataEncoder(),
+                0);
+    }
+
+    /**
+     * Loads model data from path.
+     *
+     * @param env Stream execution environment.
+     * @param path Model path.
+     * @return Knn model.
+     */
+    public static KnnModel load(StreamExecutionEnvironment env, String path) throws IOException {
+        KnnModel model = ReadWriteUtils.loadStageParam(path);
+        Table modelDataTable = ReadWriteUtils.loadModelData(env, path, new ModelDataDecoder(), 0);
+        return model.setModelData(modelDataTable);
+    }
+
+    /** This operator loads model data and predicts result. */
+    private static class PredictLabelFunction extends RichMapFunction<Row, Row> {
+        private final String featureCol;
+        private KnnModelData knnModelData;
+        private final int k;
+        private final String broadcastKey;
+
+        public PredictLabelFunction(String broadcastKey, int k, String featureCol) {
+            this.k = k;
+            this.broadcastKey = broadcastKey;
+            this.featureCol = featureCol;
+        }
+
+        @Override
+        public Row map(Row row) {
+            if (knnModelData == null) {
+                knnModelData =
+                        (KnnModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastKey).get(0);
+            }
+            DenseVector vector = (DenseVector) row.getField(featureCol);
+            if (vector == null) {
+                return Row.join(row, new Row(1));
+            }
+            Tuple2<List<Double>, List<Double>> tuple2 = findNeighbor(vector);
+            double percent = 1.0 / tuple2.f0.size();
+            Map<Double, Double> detail = new HashMap<>(0);
+            for (Double obj : tuple2.f0) {
+                detail.merge(obj, percent, Double::sum);
+            }
+            double max = 0.0;
+            double prediction = 0.0;
+            for (Map.Entry<Double, Double> entry : detail.entrySet()) {
+                if (entry.getValue() > max) {
+                    max = entry.getValue();
+                    prediction = entry.getKey();
+                }
+            }
+            return Row.join(row, Row.of(prediction));
+        }
+
+        /** Finds the nearest k neighbors from whole vectors in matrix. */
+        private Tuple2<List<Double>, List<Double>> findNeighbor(DenseVector input) {
+            PriorityQueue<Tuple2<Double, Double>> priorityQueue =
```

Comment:
nits: would it be slightly better to rename this as `nearestKNeighbors`?

Reply:
OK

Reply:
I also believe `nearestKNeighbors` is a better name. Can you also make the renaming here?

Reply:
I think predictLabel() is better. Thanks for you suggestion.